cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -351,6 +368,8 @@ public synchronized void updateProperties(NamespaceId namespaceId, NamespaceMeta
                                                     "is created.", difference, namespaceId));
     }
     NamespaceMeta updatedMeta = builder.build();
+    // check whether the updated namespace meta still has valid Kerberos configuration
+    validateKerberosConfig(updatedMeta);
```

Comment:
This will show an misleading error message if keytab uri is provided but the existing metadata doesn't have principal. It's better to check above if the existing meta has principal and keytab before accepting a update request that has the keytab.

Reply:
right, actually no need to perform this check here, since such check is already done when construction the `builder`.